src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java

```diff
@@ -629,35 +650,34 @@ protected static String parseJavadocMemory( String memory )
             throw new IllegalArgumentException( "The memory could not be null." );
         }
 
-        Pattern p = Pattern.compile( "^\\s*(\\d+)\\s*?\\s*$" );
+        Pattern p = PARSE_JAVADOC_MEMORY_PATTERN_0;

```

Comment:
renaming this MEMORY_PATTERN_N does not help, arguably it makes this worse by making one look up the pattern

Reply:
@elharo basic idea is the original code compile the pattern every time it invoke, but actually we need compile it only one time.
after compile, it can be used thus do not need to compile it again when next try.